### PR TITLE
Temporarily set upper bound on compiler.

### PR DESCRIPTION
**Pull request by novemberkilo**
The build is failing on OCaml 5 due to dream. Will revert this after fixing dream to work with OCaml 5.

### CODE REVIEW

**Review by maiste — APPROVED** [review]
It works on my side: I merge as soon as there is a second approval :+1:

File: Dockerfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ RUN opam pin add -yn current_docker.dev "./ocurrent" && \
     opam pin add -yn dockerfile.dev "./ocaml-dockerfile" && \
     opam pin add -yn dockerfile-opam.dev "./ocaml-dockerfile" && \
     opam pin add -yn ocluster-api.dev "./ocluster"
-COPY --chown=opam ocaml-ci.opam ocaml-ci-service.opam ocaml-ci-api.opam ocaml-ci-solver.opam /src/
+COPY --chown=opam ocaml-ci.opam ocaml-ci-web.opam ocaml-ci-service.opam ocaml-ci-api.opam ocaml-ci-solver.opam /src/
```
Comment (tmcgilchrist) [inline Dockerfile:36]: This shouldn't include `ocaml-ci-web.opam`

Comment (novemberkilo) [reply Dockerfile:36]: Ah yep sorry that survived the final cleanup (I was trying to figure out why `test` wasn't able to find `ocaml-ci-web.representation`)

**Comment by maiste** [comment]
I will merge as soon as the CI works.

**Pull request merged**